geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/GenericProtocolServerConnection.java

```diff
@@ -72,16 +87,34 @@ protected void doOneMessage() {
       logger.warn(e);
       this.setFlagProcessMessagesAsFalse();
       setClientDisconnectedException(e);
+    } finally {
+      acceptor.getClientHealthMonitor().receivedPing(this.clientProxyMembershipID);
     }
   }
 
+  private void setClientProxyMembershipId() {
+    ServerLocation serverLocation = new ServerLocation(
+        ((InetSocketAddress) this.getSocket().getRemoteSocketAddress()).getHostName(),
+        this.getSocketPort());
+    DistributedMember distributedMember = new InternalDistributedMember(serverLocation);
+    // no handshake for new client protocol.
+    clientProxyMembershipID = new ClientProxyMembershipID(distributedMember);
+  }
+
   @Override
   protected boolean doHandShake(byte epType, int qSize) {
-    // no handshake for new client protocol.
+    getAcceptor().getClientHealthMonitor().registerClient(clientProxyMembershipID);
```

Comment:
I would rather see the client health monitor as a variable here, just because the contract doesn't seem to specify whether an `Acceptor`'s `ClientHealthMonitor` can change, and I'm paranoid.

Reply:
awful enough the ClientHealthMonitor is actually a Singleton. I'll change this anyways though to make it easier to make it not a Singleton.